test/app-luatest/http_client_test.lua

```diff
@@ -60,20 +61,22 @@ end
 
 g.before_each(function(cg)
     local sock_family = cg.params.sock_family
+    package.loaded['http.client'] = nil
```

Comment:
It's quite strange solution. Just try to use `require('http.client').new()` to create isolated client.

Reply:
Yes, it is a bit strange and unusual. However, there are at least two reasons to use it:
- we can achieve better isolation when module fully loaded 
- `http.client` has two interfaces:
  - with default client and access to table fields:
```lua
local httpc = require('http.client')
httpc.get('https://luajit.org')
```
  - with custom client
```lua
local httpc = require('http.client').new()
httpc.get('https://luajit.org')
```
and the second client's method cannot be accessed via dot:
```lua
tarantool> httpc = require('http.client').new()
---
...

tarantool> httpc.get('https://ya.ru/').status
---
- error: 'builtin/http.client.lua:628: Use client:get(...) instead of client.get(...):'
...

tarantool> 
```
so with `require('http.client').new()` most part of existed testcases will not work

So I left this as is.

Reply:
> we can achieve better isolation when module fully loaded

I don't understand that you mean when say "better". Standalone http client has enough isolation

> http.client has two interfaces: 

It's actually the same things. `httpc.get` is just a wrapper around some `static_client` that is also created via `httpc.new()`. So there is no any sense to even test it this static client deeply because it's just a test of this wrapper.